### PR TITLE
build: update starter to use latest @yext/pages@1.0.0-rc.7

### PR DESCRIPTION
**Pull request by asanehisa**
Updated this starter to use the latest pagesJS. Swapped @vitejs/plugin-react to @vitejs/plugin-react-swc and simplified package.json scripts.

Tested manually in dev + prod and everything worked as expected.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,41 +8,41 @@
...
-    "node": ">=17"
+    "node": "18 || 20"
   },
   "dependencies": {
-    "@headlessui/react": "^1.7.12",
-    "@heroicons/react": "^2.0.14",
-    "@yext/sites-components": "^1.0.0-rc.4",
+    "@headlessui/react": "^1.7.17",
+    "@heroicons/react": "^2.1.1",
+    "@yext/pages-components": "1.0.0-rc.0",
```
Comment (mkilpatrick) [inline package.json]: caret, same comments as other repo

File: package.json [hunk 2]
```diff
@@ -8,41 +8,41 @@
...
-    "typescript": "^4.8.3",
-    "vite": "^4.4.9"
+    "@typescript-eslint/eslint-plugin": "^6.15.0",
+    "@typescript-eslint/parser": "^6.15.0",
+    "@vitejs/plugin-react-swc": "^3.5.0",
+    "@yext/pages": "^1.0.0-rc.6",
+    "autoprefixer": "^10.4.16",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-react": "^7.33.2",
+    "generate-license-file": "^3.0.1",
```
Comment (mkilpatrick) [inline package.json]: remove

File: package.json [hunk 3]
```diff
@@ -8,41 +8,41 @@
...
+    "postcss": "^8.4.32",
+    "prettier": "^3.1.1",
+    "tailwindcss": "^3.4.0",
+    "typescript": "^5.3.3",
+    "vite": "4.4.9"
```
Comment (mkilpatrick) [inline package.json]: caret, 4.5.1

**Review by mkilpatrick — APPROVED** [review]

**Comment by lymarrie** [comment]
I'm seeing two issues... any idea what may be causing them?

1. `dev` mode
 - I'm seeing the following error when loading the index page (though this loads without issue on the `main` branch)
`500 Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
at ReactDOMServerRenderer.render (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:4053:17)
at ReactDOMServerRenderer.read (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29)
at Module.renderToString (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27)
at Module.render (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/common/src/template/internal/_server.js:5:35)
at getServerHtml (file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/sendAppHTML.js:20:45)
at sendAppHTML (file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/sendAppHTML.js:34:11)
at async file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/serverRenderSlugRoute.js:64:5` 
- When I comment out the [Slider component on this line](https://github.com/YextSolutions/fleet-smb-starter/blob/SUMO-5561/src/components/Carousel.tsx#L192), the page loads no problem. 
2. `prod` mode
  - I see the following in the browser console log... (seems related to this [slack thread](https://yext.slack.com/archives/CC7B2751A/p1703702919756339))
```
index-n3I-l7tq.js:28 Error: invalid document type
    at M0.calculatePageType (index-afcgzgar.js:9:25659)
    at M0.makeReporter (index-afcgzgar.js:9:25895)
    at new M0 (index-afcgzgar.js:9:25234)
    at L0 (index-afcgzgar.js:9:28647)
    at Di (index-n3I-l7tq.js:26:52748)
    at Ws (index-n3I-l7tq.js:30:8604)
    at Us (index-n3I-l7tq.js:30:932)
    at Of (index-n3I-l7tq.js:30:863)
    at it (index-n3I-l7tq.js:30:722)
    at ni (index-n3I-l7tq.js:28:10771)
  index-MlonrZxe.js:9 Uncaught (in promise) Error: invalid document type
    at Ng.calculatePageType (index-MlonrZxe.js:9:25669)
    at Ng.makeReporter (index-MlonrZxe.js:9:25905)
    at new Ng (index-MlonrZxe.js:9:25244)
    at xg (index-MlonrZxe.js:9:28657)
    at Di (index-eFgAKxsd.js:26:52748)
    at Ws (index-eFgAKxsd.js:30:8604)
    at Us (index-eFgAKxsd.js:30:932)
    at Of (index-eFgAKxsd.js:30:863)
    at it (index-eFgAKxsd.js:30:722)
    at ni (index-eFgAKxsd.js:28:10771)
```

Steps to replicate: 
- git clone https://github.com/YextSolutions/fleet-smb-starter.git
- git checkout SUMO-5561
- npm i
- npm run dev
- (or npm run prod)
- provide this account ID  - 4060901 (production)
- navigate to index page (powered by [this entity](https://www.yext.com/s/4060901/entity/edit3?entityIds=1006568737))

**Comment by mkilpatrick** [comment]
> I'm seeing two issues... any idea what may be causing them?
> 
> 1. `dev` mode
> 
> * I'm seeing the following error when loading the index page (though this loads without issue on the `main` branch)
>   `500 Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object. at ReactDOMServerRenderer.render (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:4053:17) at ReactDOMServerRenderer.read (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:3690:29) at Module.renderToString (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/react-dom/cjs/react-dom-server.node.development.js:4298:27) at Module.render (/Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/common/src/template/internal/_server.js:5:35) at getServerHtml (file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/sendAppHTML.js:20:45) at sendAppHTML (file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/sendAppHTML.js:34:11) at async file:///Users/lmarrie/code/starters/fleet-smb-starter/node_modules/@yext/pages/dist/dev/server/middleware/serverRenderSlugRoute.js:64:5`
> * When I comment out the [Slider component on this line](https://github.com/YextSolutions/fleet-smb-starter/blob/SUMO-5561/src/components/Carousel.tsx#L192), the page loads no problem.
> 
> 2. `prod` mode
> 
> * I see the following in the browser console log... (seems related to this [slack thread](https://yext.slack.com/archives/CC7B2751A/p1703702919756339))
> 
> ```
> index-n3I-l7tq.js:28 Error: invalid document type
>     at M0.calculatePageType (index-afcgzgar.js:9:25659)
>     at M0.makeReporter (index-afcgzgar.js:9:25895)
>     at new M0 (index-afcgzgar.js:9:25234)
>     at L0 (index-afcgzgar.js:9:28647)
>     at Di (index-n3I-l7tq.js:26:52748)
>     at Ws (index-n3I-l7tq.js:30:8604)
>     at Us (index-n3I-l7tq.js:30:932)
>     at Of (index-n3I-l7tq.js:30:863)
>     at it (index-n3I-l7tq.js:30:722)
>     at ni (index-n3I-l7tq.js:28:10771)
>   index-MlonrZxe.js:9 Uncaught (in promise) Error: invalid document type
>     at Ng.calculatePageType (index-MlonrZxe.js:9:25669)
>     at Ng.makeReporter (index-MlonrZxe.js:9:25905)
>     at new Ng (index-MlonrZxe.js:9:25244)
>     at xg (index-MlonrZxe.js:9:28657)
>     at Di (index-eFgAKxsd.js:26:52748)
>     at Ws (index-eFgAKxsd.js:30:8604)
>     at Us (index-eFgAKxsd.js:30:932)
>     at Of (index-eFgAKxsd.js:30:863)
>     at it (index-eFgAKxsd.js:30:722)
>     at ni (index-eFgAKxsd.js:28:10771)
> ```
> 
> Steps to replicate:
> 
> * git clone https://github.com/YextSolutions/fleet-smb-starter.git
> * git checkout SUMO-5561
> * npm i
> * npm run dev
> * (or npm run prod)
> * provide this account ID  - 4060901 (production)
> * navigate to index page (powered by [this entity](https://www.yext.com/s/4060901/entity/edit3?entityIds=1006568737))

invalid document type is due to config.yaml and the AnalyticsProvider. I actually started a thread about this today with Spruce: https://yext.slack.com/archives/CC7B2751A/p1703702919756339

**Comment by lymarrie** [comment]
Regarding the first issue... the error goes away when I revert back to rc.4 (even if I keep everything else in my package the same).

**Comment by asanehisa** [comment]
Re: first issue
Should work now. Looks like it's a known problem with react-slick 
https://github.com/akiran/react-slick/issues/2206

I don't see the second problem (I have Yext CLI Version: 0.1_390)

**Comment by mkilpatrick** [comment]
Change .env file to .env.local and gitignore env

**Review by mkilpatrick — APPROVED** [review]

**Comment by lymarrie** [comment]
@mkilpatrick @asanehisa I'm unfortunately still seeing the npm run prod issue when previewing in http://localhost8000. 

Local dev works 👍  - but still seeing the invalid document type error in my console when loading the page after the prod command.

image

```
index-n3I-l7tq.js:28 Error: invalid document type
    at D0.calculatePageType (index-3kPwqjc4.js:9:25659)
    at D0.makeReporter (index-3kPwqjc4.js:9:25895)
    at new D0 (index-3kPwqjc4.js:9:25234)
    at F0 (index-3kPwqjc4.js:9:28647)
    at Di (index-n3I-l7tq.js:26:52748)
    at Ws (index-n3I-l7tq.js:30:8604)
    at Us (index-n3I-l7tq.js:30:932)
    at Of (index-n3I-l7tq.js:30:863)
    at it (index-n3I-l7tq.js:30:722)
    at ni (index-n3I-l7tq.js:28:10771)
```

**Comment by mkilpatrick** [comment]
> @mkilpatrick @asanehisa I'm unfortunately still seeing the npm run prod issue when previewing in http://localhost8000.
> 
> Local dev works 👍 - but still seeing the invalid document type error in my console when loading the page after the prod command.
> 
> image
> ```
> index-n3I-l7tq.js:28 Error: invalid document type
>     at D0.calculatePageType (index-3kPwqjc4.js:9:25659)
>     at D0.makeReporter (index-3kPwqjc4.js:9:25895)
>     at new D0 (index-3kPwqjc4.js:9:25234)
>     at F0 (index-3kPwqjc4.js:9:28647)
>     at Di (index-n3I-l7tq.js:26:52748)
>     at Ws (index-n3I-l7tq.js:30:8604)
>     at Us (index-n3I-l7tq.js:30:932)
>     at Of (index-n3I-l7tq.js:30:863)
>     at it (index-n3I-l7tq.js:30:722)
>     at ni (index-n3I-l7tq.js:28:10771)
> ```

Update your Yext CLI version

**Pull request merged**